src/Orleans/Runtime/IGrainActivationContext.cs

```diff
@@ -24,5 +25,9 @@ public interface IGrainActivationContext
 
         /// <summary>Gets a key/value collection that can be used to share data within the scope of the grain activation.</summary>
         IDictionary<object, object> Items { get; }
+
+        IGrainLifeCycle ObservableLifeCycle { get; }
+
+        ParameterInfo[] ConstructorParameters { get; }
```

Comment:
Since that data is internal to every container and the abstraction/default implementation does not give us a way to access this information it is "impossible" to get.

What we maybe can do is that since our default chosen container implementation is the Asp.Net Core DI container, we can use the internal ```ConstructorMatcher``` as source code to make sure we're selecting the same constructor what the container would use. BUT if the user is configuring a different container, then our default activator would not be usable or just to some extent with some constraint, but the serviceprovider based one would work, IF the configured conforming container supports retrieving metadata from ctor parameters.